PostToExternalServices.py

```diff
@@ -0,0 +1,156 @@
+import json
+import logging
+import os
+
+import boto3
+import requests
+import tweepy
+
+AUTH0_BASE = "https://calligre.auth0.com"
+AUTH0_API = "%s/api/v2/" % AUTH0_BASE
+FB_BASE = "https://graph.facebook.com/v2.7"
+S3_BUCKET = "calligre-media"
+S3_CONFIG_BUCKET = "calligre-config"
+
+s3_client = boto3.client('s3')  # pylint: disable=C0103
+
+secrets = {}  # pylint: disable=C0103
+
+log = logging.getLogger()  # pylint: disable=C0103
+log.setLevel(logging.DEBUG)
+
+
+def set_secrets():
+    log.info("Fetching secrets from S3")
+    try:
+        response = s3_client.get_object(
+            Bucket=S3_CONFIG_BUCKET,
+            Key="secrets.json",
+        )
+    except Exception as ex:
+        log.exception(ex)
+        raise ex
+
+    try:
+        content = response.get("Body").read()
+    except AttributeError as ex:
+        log.error("S3 response didn't contain Body: %s", response)
+        log.exception(ex)
+    retrieved_secrets = json.loads(content)
+    assert len(retrieved_secrets) > 0, "Didn't fetch any secrets!"
+
+    if len(retrieved_secrets) < 4:
+        log.warning("Number of secrets seems oddly small...")
+    secrets.update(retrieved_secrets)
+
+
+def set_auth0_token():
+    log.info("Fetching Auth0 token")
+    assert (secrets.get("AUTH0_CLIENT_ID") and
+            secrets.get("AUTH0_CLIENT_SECRET")),\
+        "Auth0 Client Tokens not set!"
+
+    payload = {
+        "client_id": secrets.get("AUTH0_CLIENT_ID"),
+        "client_secret": secrets.get("AUTH0_CLIENT_SECRET"),
+        "audience": AUTH0_API,
+        "grant_type": "client_credentials"
+    }
+    try:
+        token = requests.post("%s/oauth/token" % AUTH0_BASE, json=payload).\
```

Comment:
Huh, that's weird. I would ahve assumed their `/oauth/token` route was mounted on their api rather than their base domain. My bad.